src/mumble/AudioPreprocessor.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2024 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_MUMBLE_AUDIOPREPROCESSOR_H_
+#define MUMBLE_MUMBLE_AUDIOPREPROCESSOR_H_
+
+#include <cstdint>
+#include <vector>
+
+struct SpeexEchoState_;
+struct SpeexPreprocessState_;
+
+class AudioPreprocessor {
+public:
+	/**
+	 *
+	 * \return Whether the object is initialized and its methods can be safely called.
+	 */
+	constexpr explicit operator bool() const { return m_handle != nullptr; }
+
+	AudioPreprocessor();
+	AudioPreprocessor(AudioPreprocessor &&other);
```

Comment:
We probably want to delete the copy ctor and the copy-assignment operator. Additionally, we should explicitly implement move-assignment.

Reply:
The copy constructor and copy-assignment operator are automatically deleted (or, more precisely, not implicitly created) when a move constructor is defined.

As for the move-assignment operator: good idea.

Reply:
I wild much prefer being explicit about non-existing copy ctor and copy assignment operator